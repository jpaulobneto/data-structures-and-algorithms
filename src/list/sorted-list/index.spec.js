const List = require('.');

describe('SortedList', () => {
  it('can be instantiated', () => {
    expect(() => new List()).not.toThrow();
  });

  describe('add(value)', () => {
    let list;

    beforeEach(() => {
      list = new List();
    });

    it('add a value to an empty list', () => {
      list.add(1);

      expect(list.getValues()).toMatchSnapshot();
    });

    it('add an values sorted', () => {
      list.add(2);
      list.add(1);
      list.add(3);

      expect(list.getValues()).toMatchSnapshot();
    });
  });
});

