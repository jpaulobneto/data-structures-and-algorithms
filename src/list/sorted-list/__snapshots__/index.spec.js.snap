// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortedList add(value) add a value to an empty list 1`] = `
Array [
  1,
]
`;

exports[`SortedList add(value) add an values sorted 1`] = `
Array [
  1,
  2,
  3,
]
`;
